content/blog/2018-08-14-where-to-bird.md

```diff
@@ -0,0 +1,383 @@
+
+``` r
+t1 <- Sys.time()
+d <- geodist::geodist (xy_W, xy_N)
+# so fast!!!
+Sys.time() - t1
+```
+
+    ## Time difference of 13.57983 secs
+
+``` r
+d1 <- apply (d, 1, min)
+d2 <- apply (d, 2, min)
+xy <- cbind (rbind (xy_W, xy_N), c (d1, d2))
+xy <- xy [, c (1, 2, 5)]
+colnames (xy) <- c ("x", "y", "z")
+xy <- xy [!duplicated (xy), ]
+```
+
+Finally we plot the results on the roads we had gotten earlier, although
+we do not recommend staying on the middle of a road to observe birds! We
+re-add the points corresponding to the MPI and bird hide after the
+surface coloring. With `osmplotr`, order matters because layers are
+added on top of each other. Note that plotting the distances takes a
+while.
+
+``` r
+# colorblind-friendly palette!
+cols <- viridis::viridis_pal (direction = -1) (30)
+
+add_osm_surface (map0, dat_H,
+                 dat = xy, col = cols) %>%
+    add_axes()  %>%
+  add_colourbar(cols = cols,
+                zlim = range(xy[,"z"])) %>%
+  add_osm_objects(mpi_sf, 
+                  col = 'white', size = 5) %>%
+  add_osm_objects(results$osm_points, 
+                  col = 'white', size = 5)%>%
+  print_osm_map (filename = 'map_a3.png', width = 600,
+                 units = 'px', dpi = 72)
+
+magick::image_read('map_a3.png') %>%
+  magick::image_annotate("Map data © OpenStreetMap contributors",
+                         color = "white",
+                         boxcolor =  "black",
+                         size = 15,
+                         gravity = "south")
+```
+
+<img src="/img/blog-images/2018-08-14-where-to-bird/unnamed-chunk-9-1.png" width="600" />
+
+On the map, the yellower/lighter a road is, the better it is to observe
+birds according to Mark’s assumption that birds are most likely to be
+found where water lies close to natural areas. With this metric, the MPI
+itself is not too badly located, which after all is not too surprising
+for an MPI of *ornithology*. Maybe one should just walk to the one of
+the nearest lakes to meet some birds.
+
+### Conclusion
+
+#### A note about open geographical data
+
+In this post we used `opencage`, `osmdata` and `osmplotr` therefore
+mosly making use of the awesome OpenStreetMap data (The OpenCage
+Geocoder uses a bit more, but it only warrants [attributing
+OpenStreetMap](https://opencagedata.com/faq#legal)). We therefore added
+the annotation “Map data © OpenStreetMap contributors” to all maps of
+this post using rOpenSci’s `magick` package. You might also have noticed
+in the code earlier that both `osmdata` and `osmplotr` have start-up
+messages indicating the data origin and licence.
+
+Another package of rOpenSci’s interacting with open geographical data,
+that might be of interest for making maps, is
+[`rnaturalearth`](https://github.com/ropensci/rnaturalearth) that
+facilitates interaction with [Natural Earth map
+data](http://www.naturalearthdata.com/).
+
+#### Geographical analyses with R
+
+In this post we showcased three rOpenSci packages helping you use
```

Comment:
I just notice that some sentences in "Geographical analyses with R" are so similar to some in "A note about open geographical data" that I stopped reading to see if they were duplicated.

> In this post we used opencage, osmdata and osmplotr ...

> In this post we showcased ... opencage, osmdata, osmplotr. 

and `magick` is noted twice. I understand why you tried separate paragraphs but it feels like they really could be merged to " A note about open geographical data and analyses with R" also because the last section of the data para also talks about rOpenSci packages. For the reader the difference between the two paras is too subtle so therefore confusing.

Reply:
You might start the merged para with "In this post we mostly made use of the awesome OpenStreetMap data ...", note licences in startup messages and then go into the packages that access that data and magick to annotate images and then `rnaturalearth`. Next para start "We also used the geodist package..."